pkg/apis/controlplane/v1beta2/types.go

```diff
@@ -109,6 +109,21 @@ type ClusterGroupMembers struct {
 	CurrentPage       int64         `json:"currentPage" protobuf:"varint,6,opt,name=currentPage"`
 }
 
+// +genclient
+// +genclient:onlyVerbs=get
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GroupMembers is a list of GroupMember objects or IPBlocks that are currently selected by a Group.
+type GroupMembers struct {
```

Comment:
Same question - why we need to redefine a struct?

Reply:
I redefined because on line 97 `// +genclient:nonNamespaced`, ClusterGroupMembers API is non namespaced and GroupMembers API is namespaced, and the `genclient` is auto generated. So I only generalized the pagination function not the Get(). Is there any suggestion to solve the problem? I'm not aware of, thanks!

Reply:
Could we do:
`type ClusterGroupMember GroupMember`?

Reply:
Here? Then they are still not differentiated in terms of `nonNamespaced` and `namespaced` 🤔